### PR TITLE
chore(common): CHECKOUT-6377 Setup apple pay package

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
Setup apple pay package and move apple pay payment strategy

## Why?
As part of monorepo setup move apple pay to a separate package.

## Testing / Proof
- circle

@bigcommerce/checkout

### CODE REVIEW

File: packages/apple-pay/create-apple-pay-payment-strategy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { PaymentIntegrationService, PaymentIntegrationServiceMock } from "@bigcommerce/checkout-sdk/payment-integration";
```
Comment (davidchin) [inline packages/apple-pay/create-apple-pay-payment-strategy.spec.ts:1]: Should this file live inside `src` next to the file it is testing against?

File: packages/payment-integration/src/test-utils/line-items.mock.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { DigitalItem, GiftCertificateItem, PhysicalItem } from "../cart/line-item";
```
Comment (davidchin) [inline packages/payment-integration/src/test-utils/line-items.mock.ts:1]: We should be using single quotes, but I believe it'll be once we apply `@bigcommerce/eslint-config` to the project.

Comment (animesh1987) [reply packages/payment-integration/src/test-utils/line-items.mock.ts:1]: Yes agreed, I think this is my IDE doing things.

File: packages/apple-pay/create-apple-pay-payment-strategy.spec.ts (hunk 1 above)
Comment (animesh1987) [reply packages/apple-pay/create-apple-pay-payment-strategy.spec.ts]: Good catch. It was created there by accident. Updated

File: packages/payment-integration/src/test-utils/line-items.mock.ts (hunk 2 above)
Comment (animesh1987) [reply packages/payment-integration/src/test-utils/line-items.mock.ts:1]: This will move it as part of the `@bigcommerce/eslint-config` migration

**Review by davidchin — APPROVED** [review]

**Pull request merged**